openshift-serverless/tasks/main.yml

```diff
@@ -0,0 +1,204 @@
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "Ready"
+    wait_timeout: 90
+- name: Get hello service Conditions "RoutesReady"
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: hello
+    api_version: serving.knative.dev/v1
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "RoutesReady"
+    wait_timeout: 90
+  register: service
+  failed_when: 
+    - (service.resources[0].status.conditions | selectattr('type', 'search', 'Ready') | list | last).status != "True"
+    - (service.resources[0].status.conditions | selectattr('type', 'search', 'RoutesReady') | list | last).status != "True"
+- name: Check hello service - must return hello serverless
+  ansible.builtin.uri:
+    url: "{{service.resources[0].status.url}}"
+    return_content: true
+    validate_certs: false
+  register: hello_msg
+  failed_when: "'Hello Serverless!' not in hello_msg.content"
+# 4. Test auto-scale service
+- name: Create knative hello-autoscale service
+  kubernetes.core.k8s:
+    api_version: serving.knative.dev/v1
+    kind: Service
+    name: hello-autoscale
+    namespace: knative-serving
+    definition:
+      spec:
+        template:
+          metadata:
+            annotations:
+              autoscaling.knative.dev/min-scale: "0"
+              autoscaling.knative.dev/max-scale: "20"
+          spec:
+            containers:
+              - image: quay.io/jianrzha/go-http-server:1.0.0
+                env:
+                  - name: RESPONSE
+                    value: "Hello Serverless!"
+- name: Get hello-autoscale service Conditions "Ready"
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: hello-autoscale
+    api_version: serving.knative.dev/v1
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "Ready"
+    wait_timeout: 90
+- name: Get hello-autoscale service Conditions "RoutesReady"
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: hello-autoscale
+    api_version: serving.knative.dev/v1
+    kind: Service
+    wait: yes
+    wait_condition:
+      status: "True"
+      type: "RoutesReady"
+    wait_timeout: 90
+  register: service_autoscale
+  failed_when: 
+    - (service_autoscale.resources[0].status.conditions | selectattr('type', 'search', 'Ready') | list | last).status != "True"
+    - (service_autoscale.resources[0].status.conditions | selectattr('type', 'search', 'RoutesReady') | list | last).status != "True"
+- name: Check hello service autoscale - must return Hello world from sample HTTP server
+  ansible.builtin.uri:
+    url: "{{service_autoscale.resources[0].status.url}}"
+    return_content: true
+    validate_certs: false
+  register: autoscale_hello_msg
+  failed_when: "'Hello world' not in autoscale_hello_msg.content"
+# If it didn't failed till here all tests must have succeeded
+- name: Openshift-Serverless results
+  debug:
```

Comment:
Always use the FQCN

`ansible.builtin.debug`

Reply:
done